src/huggingface_hub/hf_api.py

```diff
@@ -1148,6 +1148,34 @@ def list_metrics(self) -> List[MetricInfo]:
         d = r.json()
         return [MetricInfo(**x) for x in d]
 
+    def list_spaces(self, space_sdk: Optional[str] = None) -> List[SpaceInfo]:
```

Comment:
Consider using parameters similar to the ones in https://github.com/huggingface/huggingface_hub/blob/main/src/huggingface_hub/hf_api.py#L763-L776. I think here it would be nice to have `use_auth_token` so people can filter for private Spaces.